tools/lsp/server_loop.rs

```diff
@@ -158,6 +69,7 @@ fn create_show_preview_command(
     )
 }
 
+// A cache of loaded documents
```

Comment:
Add documentation comments when it makes sense 

```suggestion
/// A cache of loaded documents
```